opentracing-util/src/main/java/io/opentracing/util/AutoFinisher.java

```diff
@@ -0,0 +1,53 @@
+package io.opentracing.util;
+
+import io.opentracing.ActiveSpan;
+import io.opentracing.Span;
+import io.opentracing.SpanContext;
+
+import java.util.Map;
+import java.util.concurrent.atomic.AtomicInteger;
+
+/**
+ * XXX: fix comment
+ * {@link AutoFinisher} is an {@link ActiveSpan} wrapper that automatically {@link Span#finish()}es the underlying
+ * {@link Span} when there are zero remaining {@link ActiveSpan}s or {@link ActiveSpan.Continuation}s referencing
+ * that underlying {@link Span}.
+ *
+ * <p>
+ * Use {@link AutoFinisher} like this:
+ * <pre><code>
+ *     try (ActiveSpan span = new AutoFinisher(tracer.buildSpan("...").startActive())) {
```

Comment:
example doesn't match the no-arg constructor `AutoFinisher()`.

The trick with AutoFinisher is that there can be only one per Span, otherwise ref-counting will be broken